command/command.go

```diff
@@ -9,121 +9,143 @@ import (
 	"strings"
 )
 
-// ----------
-
-// Model ...
-type Model struct {
+// Command ...
+type Command interface {
+	SetDir(dir string) Command
```

Comment:
Why do we keep this builder type configuration that seems kinda antipattern in Go instead of adding these to the constructor and factory method?

Reply:
Hmm thinking more maybe it is worth keeping the builder for these "optional" type of arguments so we avoid polluting the factory methods / constructors. (No default and named values in golang luckily...)

Reply:
Yeah, I think until we don't have default parameters AND method overloading, adding a new constructor for all combinations would be very painful.

Reply:
Discussed above ☝️